### PR TITLE
fix: launch_training.py arguments with new tracker api

### PR DESCRIPTION
**Pull request by dushyantbehl**
Fixes launch training according to the new tracker api.

### Was the PR tested

- [ ] I have added >=1 unit test(s) for every new method I have added.
- [ ] I have ensured all unit tests pass

### CODE REVIEW

File: build/utils.py [hunk 1]
```diff
@@ -25,7 +25,13 @@
...
 
 # Local
-from tuning.config import configs, peft_config
+from tuning.config import configs, peft_config, tracker_configs
+
+# from tuning.config. import (
+#    FileLoggingTrackerConfig,
+#    AimConfig,
+#    TrackerConfigFactory
+# )
```
Comment (kmehant) [inline build/utils.py:34]: Shall we remove this?

Comment (dushyantbehl) [reply build/utils.py]: removed

**Comment by dushyantbehl** [comment]
I tested this PR with these steps,

Followed the steps listed here for [image creation](https://github.com/foundation-model-stack/fms-hf-tuning/wiki/Installing-and-Testing-OpenShift-fms%E2%80%90hf%E2%80%90tuning-Stack#manually-building-and-pushing-an-image-to-the-openshift-cluster-repository) and ran the test

```
cat <<EOF | oc apply -f -
apiVersion: v1
kind: ConfigMap
metadata:
  name: my-config
  namespace: default
data:
  config.json: |
    {
      "accelerate_launch_args": {
        "num_machines": 1,
        "num_processes": 2
      },
      "model_name_or_path": "bigscience/bloom-560m",
      "training_data_path": "/etc/config/twitter_complaints_small.json",
      "output_dir": "/tmp/out",
      "num_train_epochs": 1.0,
      "per_device_train_batch_size": 4,
      "per_device_eval_batch_size": 4,
      "gradient_accumulation_steps": 4,
      "evaluation_strategy": "no",
      "save_strategy": "epoch",
      "learning_rate": 1e-5,
      "weight_decay": 0.0,
      "lr_scheduler_type": "cosine",
      "logging_steps": 1.0,
      "packing": false,
      "include_tokens_per_second": true,
      "response_template": "\n### Label:",
      "dataset_text_field": "output",
      "use_flash_attn": false,
      "torch_dtype": "float32",
      "peft_method": "pt",
      "tokenizer_name_or_path": "bigscience/bloom"
    }
  twitter_complaints_small.json: |
    {"Tweet text":"@HMRCcustomers No this is my first job","ID":0,"Label":2,"text_label":"no complaint","output":"### Text: @HMRCcustomers No this is my first job\n\n### Label: no complaint"}
    {"Tweet text":"@KristaMariePark Thank you for your interest! If you decide to cancel, you can call Customer Care at 1-800-NYTIMES.","ID":1,"Label":2,"text_label":"no complaint","output":"### Text: @KristaMariePark Thank you for your interest! If you decide to cancel, you can call Customer Care at 1-800-NYTIMES.\n\n### Label: no complaint"}
    {"Tweet text":"If I can't get my 3rd pair of @beatsbydre powerbeats to work today I'm doneski man. This is a slap in my balls. Your next @Bose @BoseService","ID":2,"Label":1,"text_label":"complaint","output":"### Text: If I can't get my 3rd pair of @beatsbydre powerbeats to work today I'm doneski man. This is a slap in my balls. Your next @Bose @BoseService\n\n### Label: complaint"}
    {"Tweet text":"@EE On Rosneath Arial having good upload and download speeds but terrible latency 200ms. Why is this.","ID":3,"Label":1,"text_label":"complaint","output":"### Text: @EE On Rosneath Arial having good upload and download speeds but terrible latency 200ms. Why is this.\n\n### Label: complaint"}
    {"Tweet text":"Couples wallpaper, so cute. :) #BrothersAtHome","ID":4,"Label":2,"text_label":"no complaint","output":"### Text: Couples wallpaper, so cute. :) #BrothersAtHome\n\n### Label: no complaint"}
    {"Tweet text":"@mckelldogs This might just be me, but-- eyedrops? Artificial tears are so useful when you're sleep-deprived and sp\u2026 https:\/\/t.co\/WRtNsokblG","ID":5,"Label":2,"text_label":"no complaint","output":"### Text: @mckelldogs This might just be me, but-- eyedrops? Artificial tears are so useful when you're sleep-deprived and sp\u2026 https:\/\/t.co\/WRtNsokblG\n\n### Label: no complaint"}
    {"Tweet text":"@Yelp can we get the exact calculations for a business rating (for example if its 4 stars but actually 4.2) or do we use a 3rd party site?","ID":6,"Label":2,"text_label":"no complaint","output":"### Text: @Yelp can we get the exact calculations for a business rating (for example if its 4 stars but actually 4.2) or do we use a 3rd party site?\n\n### Label: no complaint"}
    {"Tweet text":"@nationalgridus I have no water and the bill is current and paid. Can you do something about this?","ID":7,"Label":1,"text_label":"complaint","output":"### Text: @nationalgridus I have no water and the bill is current and paid. Can you do something about this?\n\n### Label: complaint"}
    {"Tweet text":"Never shopping at @MACcosmetics again. Every time I go in there, their employees are super rude\/condescending. I'll take my $$ to @Sephora","ID":8,"Label":1,"text_label":"complaint","output":"### Text: Never shopping at @MACcosmetics again. Every time I go in there, their employees are super rude\/condescending. I'll take my $$ to @Sephora\n\n### Label: complaint"}
    {"Tweet text":"@JenniferTilly Merry Christmas to as well. You get more stunning every year \ufffd\ufffd","ID":9,"Label":2,"text_label":"no complaint","output":"### Text: @JenniferTilly Merry Christmas to as well. You get more stunning every year \ufffd\ufffd\n\n### Label: no complaint"}
---
apiVersion: "kubeflow.org/v1"
kind: PyTorchJob
metadata:
  name: ted-kfto-sft
  namespace: default
  labels:
    kueue.x-k8s.io/queue-name: lq-trainer
spec:
  pytorchReplicaSpecs:
    Master:
      replicas: 1
      restartPolicy: Never # Do not restart the pod on failure. If you do set it to OnFailure, be sure to also set backoffLimit
      template:
        spec:
          containers:
            - name: pytorch
              # This is the temp location util image is officially released
              image: image-registry.openshift-image-registry.svc:5000/opendatahub/fms-hf-tuning:db-fix-build-failure
              imagePullPolicy: IfNotPresent
              command:
                - "python"
                - "/app/accelerate_launch.py"
              env:
                - name: SFT_TRAINER_CONFIG_JSON_PATH
                  value: /etc/config/config.json
              volumeMounts:
              - name: config-volume
                mountPath: /etc/config
          volumes:
          - name: config-volume
            configMap:
              name: my-config
              items:
              - key: config.json
                path: config.json
              - key: twitter_complaints_small.json
                path: twitter_complaints_small.json
EOF
```

This is the output,

```
➜  fms-hf-tuning git:(fix-pr89-launch-training) oc get pytorchjobs,pods -n default          
NAME                                   STATE     AGE
pytorchjob.kubeflow.org/ted-kfto-sft   Created   94s

NAME                        READY   STATUS              RESTARTS   AGE
pod/ted-kfto-sft-master-0   0/1     ContainerCreating   0          95s
➜  fms-hf-tuning git:(fix-pr89-launch-training) oc get pytorchjobs,pods -n default
NAME                                   STATE     AGE
pytorchjob.kubeflow.org/ted-kfto-sft   Running   3m2s

NAME                        READY   STATUS    RESTARTS   AGE
pod/ted-kfto-sft-master-0   1/1     Running   0          3m2s
➜  fms-hf-tuning git:(fix-pr89-launch-training) oc logs -f pod/ted-kfto-sft-master-0 -n default
WARNING:root:SET_NUM_PROCESSES_TO_NUM_GPUS=True, overwriting user set num_processes 2                to all GPUs available, 0.
WARNING:root:num_processes param was not passed in. Value from config file (if available) will                 be used or accelerate launch will determine number of processes automatically
WARNING:accelerate.commands.launch:The following values were not passed to `accelerate launch` and had defaults used instead:
	`--num_processes` was set to a value of `0`
	`--mixed_precision` was set to a value of `'no'`
	`--dynamo_backend` was set to a value of `'no'`
To avoid this warning pass in values for each of the problematic parameters or run `accelerate config`.
/usr/local/lib/python3.11/site-packages/huggingface_hub/file_download.py:1132: FutureWarning: `resume_download` is deprecated and will be removed in version 1.0.0. Downloads always resume when possible. If you want to force a new download, use `force_download=True`.
  warnings.warn(
Generating train split: 10 examples [00:00, 2791.92 examples/s]
Map: 100%|██████████| 10/10 [00:00<00:00, 1704.38 examples/s]
Map: 100%|██████████| 10/10 [00:00<00:00, 1375.68 examples/s]
/usr/local/lib/python3.11/site-packages/trl/trainer/sft_trainer.py:318: UserWarning: You passed a tokenizer with `padding_side` not equal to `right` to the SFTTrainer. This might lead to some unexpected behaviour due to overflow issues when training a model in half-precision. You might consider adding `tokenizer.padding_side = 'right'` to your code.
  warnings.warn(
100%|██████████| 1/1 [00:25<00:00, 25.99s/it]{'loss': 4.8113, 'grad_norm': 17.683847427368164, 'learning_rate': 0.0, 'epoch': 1.0}
100%|██████████| 1/1 [00:25<00:00, 25.99s/it]/usr/local/lib/python3.11/site-packages/peft/utils/save_and_load.py:168: UserWarning: Setting `save_embedding_layers` to `True` as the embedding layer has been resized during finetuning.
  warnings.warn(
{'train_runtime': 30.325, 'train_samples_per_second': 0.33, 'train_steps_per_second': 0.033, 'train_tokens_per_second': 15.367, 'train_loss': 4.811284065246582, 'epoch': 1.0}
100%|██████████| 1/1 [00:30<00:00, 30.32s/it]
➜  fms-hf-tuning git:(fix-pr89-launch-training) oc get pytorchjobs,pods -n default             
NAME                                   STATE       AGE
pytorchjob.kubeflow.org/ted-kfto-sft   Succeeded   3m43s

NAME                        READY   STATUS      RESTARTS   AGE
pod/ted-kfto-sft-master-0   0/1     Completed   0          3m43s
```
If we need any other tests, let me know else the PR is ready.

**Comment by kmehant** [comment]
@dushyantbehl can you fix the lint issues? Thank you.

**Comment by dushyantbehl** [comment]
@kmehant fixed

**Comment by dushyantbehl** [comment]
Also verified that model and logs are present in the directory `/tmp/out` for this test 

```
➜  fms-hf-tuning git:(fix-pr89-launch-training) oc rsh ted-kfto-sft-master-0                                                                                                                                                                                  
sh-5.1$ bash
[tuning@ted-kfto-sft-master-0 app]$ ls
accelerate_fsdp_defaults.yaml  accelerate_launch.py  build  launch_training.py
[tuning@ted-kfto-sft-master-0 app]$ python /app/accelerate_launch.py
WARNING:root:SET_NUM_PROCESSES_TO_NUM_GPUS=True, overwriting user set num_processes 2                to all GPUs available, 0.
WARNING:root:num_processes param was not passed in. Value from config file (if available) will                 be used or accelerate launch will determine number of processes automatically
WARNING:accelerate.commands.launch:The following values were not passed to `accelerate launch` and had defaults used instead:
	`--num_processes` was set to a value of `0`
	`--mixed_precision` was set to a value of `'no'`
	`--dynamo_backend` was set to a value of `'no'`
To avoid this warning pass in values for each of the problematic parameters or run `accelerate config`.
/usr/local/lib/python3.11/site-packages/huggingface_hub/file_download.py:1132: FutureWarning: `resume_download` is deprecated and will be removed in version 1.0.0. Downloads always resume when possible. If you want to force a new download, use `force_download=True`.
  warnings.warn(
config.json: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 693/693 [00:00<00:00, 4.82MB/s]
model.safetensors: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1.12G/1.12G [00:15<00:00, 74.4MB/s]
tokenizer_config.json: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 222/222 [00:00<00:00, 2.06MB/s]
tokenizer.json: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 14.5M/14.5M [00:00<00:00, 76.0MB/s]
special_tokens_map.json: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 85.0/85.0 [00:00<00:00, 377kB/s]
Generating train split: 10 examples [00:00, 2857.35 examples/s]
Map: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00<00:00, 1482.87 examples/s]
tokenizer_config.json: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 222/222 [00:00<00:00, 1.52MB/s]
tokenizer.json: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 14.5M/14.5M [00:00<00:00, 31.1MB/s]
special_tokens_map.json: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 85.0/85.0 [00:00<00:00, 510kB/s]
Map: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00<00:00, 1619.36 examples/s]
/usr/local/lib/python3.11/site-packages/trl/trainer/sft_trainer.py:318: UserWarning: You passed a tokenizer with `padding_side` not equal to `right` to the SFTTrainer. This might lead to some unexpected behaviour due to overflow issues when training a model in half-precision. You might consider adding `tokenizer.padding_side = 'right'` to your code.
  warnings.warn(
{'loss': 4.8113, 'grad_norm': 17.683847427368164, 'learning_rate': 0.0, 'epoch': 1.0}                                                                                                                                                                         
100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:24<00:00, 24.99s/it]/usr/local/lib/python3.11/site-packages/peft/utils/save_and_load.py:168: UserWarning: Setting `save_embedding_layers` to `True` as the embedding layer has been resized during finetuning.
  warnings.warn(
{'train_runtime': 28.9665, 'train_samples_per_second': 0.345, 'train_steps_per_second': 0.035, 'train_tokens_per_second': 16.088, 'train_loss': 4.811284065246582, 'epoch': 1.0}                                                                              
100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:28<00:00, 28.97s/it]
[tuning@ted-kfto-sft-master-0 app]$ ls /tmp/out/
README.md  adapter_config.json	adapter_model.safetensors  optimizer.pt  rng_state.pth	scheduler.pt  special_tokens_map.json  tokenizer.json  tokenizer_config.json  trainer_state.json  training_args.bin  training_logs.jsonl
[tuning@ted-kfto-sft-master-0 app]$ 

```

**Comment by jbusche** [comment]
I tssted this PR on both a FIPS and non-FIPS cluster, and it looks good:
```
Every 2.0s: oc get pytorchjobs,pods                                                                              api.fips-test.cp.fyre.ibm.com: Fri May 10 10:01:56 2024

NAME                                   STATE	   AGE
pytorchjob.kubeflow.org/ted-kfto-sft   Succeeded   2m19s

NAME                        READY   STATUS	RESTARTS   AGE
pod/ted-kfto-sft-master-0   0/1     Completed   0          2m19s
```
This also fixes the FIPS warning messages that were mentioned in issue [776](https://github.ibm.com/ai-foundation/watson-fm-stack-tracker/issues/776)

**Review by jbusche — APPROVED** [review]
LGTM, tested on both FIPS and non-FIPS OC 4.14.21 clusters

**Pull request merged**